third_party/blink/renderer/platform/heap_observer_set.h

```diff
@@ -33,7 +33,12 @@ class PLATFORM_EXPORT HeapObserverSet {
   // Removes the given observer from this list. Does nothing if this observer is
   // not in this list.
   void RemoveObserver(ObserverType* observer) {
-    CHECK(iteration_state_ & kAllowingRemoval);
+    CHECK(iteration_state_ & kAllowingRemoval);    
+    if (recordreplay::AreEventsDisallowed() &&
+        recordreplay::IsRecordingOrReplaying("leak-references", "HeapObserverSet")) {
+      return;
```

Comment:
🤔  at "leaf-references".  this does more than leak/keep-alive: the observer stays around in the list of observers to notify.